cpp/include/cuspatial_test/vector_factories.cuh

```diff
@@ -271,22 +282,34 @@ class multilinestring_array {
  * @param coord_inl Ramge of coordinate
  * @return multilinestring array object
  */
-template <typename IndexRangeA,
-          typename IndexRangeB,
-          typename CoordRange,
-          typename IndexType = typename IndexRangeB::value_type>
-auto make_multilinestring_array(IndexRangeA geometry_inl,
-                                IndexRangeB part_inl,
-                                CoordRange coord_inl)
+template <typename IndexType, typename T>
+auto make_multilinestring_array(rmm::device_uvector<IndexType>&& geometry_inl,
+                                rmm::device_uvector<IndexType>&& part_inl,
+                                rmm::device_uvector<vec_2d<T>>&& coord_inl)
 {
-  using CoordType         = typename CoordRange::value_type;
-  using DeviceIndexVector = thrust::device_vector<IndexType>;
-  using DeviceCoordVector = thrust::device_vector<CoordType>;
+  return multilinestring_array<rmm::device_uvector<IndexType>,
+                               rmm::device_uvector<IndexType>,
+                               rmm::device_uvector<vec_2d<T>>>(
+    std::move(geometry_inl), std::move(part_inl), std::move(coord_inl));
+}
 
-  return multilinestring_array<DeviceIndexVector, DeviceIndexVector, DeviceCoordVector>(
-    make_device_vector(std::move(geometry_inl)),
-    make_device_vector(std::move(part_inl)),
-    make_device_vector(std::move(coord_inl)));
+/**
+ * @brief Construct an owning object of a multilinestring array from ranges
+ *
+ * @param geometry_inl Range of geometry offsets
+ * @param part_inl Range of part offsets
+ * @param coord_inl Ramge of coordinate
+ * @return multilinestring array object
+ */
+template <typename IndexType, typename T>
+auto make_multilinestring_array(rmm::device_vector<IndexType>&& geometry_inl,
```

Comment:
Would be nice to use spans here (in newer libcu++), or template the container type?

Reply:
Agreed, the vector factories are fairly unorganized atm. They were created adhoc as handy functions when creating test cases. Definitely needs reorganizing.

Depending on APIs, they can be promoted into the main library if a new algorithm needs to return an owning data structure to store geometries.

Reply:
I missed that this is part of tests. So it's fine to just specialize it to device_vector.